api-common/src/main/java/org/openmrs/module/metadatasharing/util/MetadataUtil.java

```diff
@@ -0,0 +1,335 @@
+		MetadataPackagesConfig config = getMetadataPackagesForModule(loader, packagesFilePath);
+		return loadPackages(config, loader);
+	}
+	
+	/**
+	 * Useful for testing, e.g. if you need to load up a specific MDS package
+	 * 
+	 * @param loader
+	 * @param namesToLoad something like Reference_Application_Visit_and_Encounter_Types
+	 * @return
+	 * @throws Exception
+	 */
+	public static boolean setupSpecificMetadata(ClassLoader loader, String... namesToLoad) throws Exception {
+		List<String> namesToKeep = Arrays.asList(namesToLoad);
+		MetadataPackagesConfig config = getMetadataPackagesForModule(loader);
+		for (Iterator<MetadataPackageConfig> i = config.getPackages().iterator(); i.hasNext();) {
+			MetadataPackageConfig packageConfig = i.next();
+			if (!namesToKeep.contains(packageConfig.getFilenameBase())) {
+				i.remove();
+			}
+		}
+		return loadPackages(config, loader);
+	}
+	
+	private synchronized static boolean loadPackages(MetadataPackagesConfig config, ClassLoader loader) throws IOException {
+		
+		List<PackageImporter> packageImporters = new ArrayList<PackageImporter>();
+		
+		// first, instantiate and load a package importer for each package
+		for (MetadataPackageConfig pkg : config.getPackages()) {
+			PackageImporter packageImporter = loadPackageImporterIfNecessary(pkg, loader);
+			if (packageImporter != null) {
+				packageImporters.add(packageImporter);
+			}
+		}
+		
+		// do the actual imports
+		if (packageImporters.size() > 0) {
+			
+			// sort in order of date created, with most recently created coming last: this is so in the case of inconsistent metadata between packages, the most recent one wins
+			Collections.sort(packageImporters, new Comparator<PackageImporter>() {
+				
+				@Override
+				public int compare(PackageImporter i1, PackageImporter i2) {
+					return i1.getImportedPackage().getDateCreated().compareTo(i2.getImportedPackage().getDateCreated());
+				}
+			});
+			
+			for (PackageImporter packageImporter : packageImporters) {
+				long timer = System.currentTimeMillis();
+				log.info("Importing package: " + packageImporter.getImportedPackage().getName());
+				packageImporter.importPackage();
+				log.info("Imported " + packageImporter.getImportedPackage().getName() + " in "
+				        + (System.currentTimeMillis() - timer) + "ms");
+			}
+			return true;
+		}
+		
+		return false;
+	}
+	
+	public static MetadataPackagesConfig getMetadataPackagesForModule(ClassLoader loader) {
+		return getMetadataPackagesForModule(loader, PACKAGES_FILENAME);
+	}
+	
+	public static MetadataPackagesConfig getMetadataPackagesForModule(ClassLoader loader, String packageFilePath) {
+		try {
+			InputStream stream = loader.getResourceAsStream(packageFilePath);
+			String xml = IOUtils.toString(stream);
+			SimpleXStreamSerializer serializer = (SimpleXStreamSerializer) Context.getSerializationService().getDefaultSerializer();

```

Comment:
Please check on this behavior here, it looks like you're setting these aliases _every time the method is called_ instead of once at setup. (I assume that Context.getSerializationService().getDefaultSerializer() doesn't generate a new serializer every time.)

Reply:
yes , i think it would generating a  new serializer if i did like  
`SimpleXStreamSerializer serializer = new SimpleXStreamSerializer().`

But my understanding of this 
`SimpleXStreamSerializer serializer = (SimpleXStreamSerializer) Context.getSerializationService().getDefaultSerializer();`

 is that i get the SimpleXStreamSerializer object once , that is te one returned by `Context.getSerializationService().getDefaultSerializer();``

Reply:
otherwise you can advise me accordingly ,